src/main/java/tds/content/mappers/ContentsItemDocumentMapper.java

```diff
@@ -0,0 +1,292 @@
+import tds.itemrenderer.data.xml.itemrelease.BrailleCode;
+import tds.itemrenderer.data.xml.itemrelease.Content;
+import tds.itemrenderer.data.xml.itemrelease.ContentLinkInfo;
+import tds.itemrenderer.data.xml.itemrelease.ItemPassage;
+import tds.itemrenderer.data.xml.itemrelease.Itemrelease;
+import tds.itemrenderer.data.xml.itemrelease.Key;
+import tds.itemrenderer.data.xml.itemrelease.Keyboard;
+import tds.itemrenderer.data.xml.itemrelease.KeyboardRow;
+import tds.itemrenderer.data.xml.itemrelease.Option;
+import tds.itemrenderer.data.xml.itemrelease.Optionlist;
+import tds.itemrenderer.data.xml.itemrelease.Qti;
+import tds.itemrenderer.data.xml.itemrelease.RelatedElementInfo;
+
+@Component
+public class ContentsItemDocumentMapper implements ItemDocumentMapper {
+    private final static String ITEMBODY_SPEC = "itemBody";
+    private final static String ITEMBODY_SPEC_ELEMENT = "<itemBody>";
+
+    @Override
+    public void mapItemDocument(final ITSDocument document, final Itemrelease itemXml) {
+        ItemPassage item = itemXml.getItemPassage();
+        for (Content content : item.getContent()) {
+            ITSContent itsContent = new ITSContent();
+            itsContent.setLanguage(content.getLanguage());
+            if (content.getQti() != null) {
+                itsContent.setQti(readQti(content.getQti()));
+            }
+            if (content.getIllustration() != null) {
+                itsContent.setIllustration(content.getIllustration());
+            }
+            if (content.getIllustrationTts() != null) {
+                itsContent.setIllustrationTTS(content.getIllustrationTts());
+            }
+            if (content.getStem() != null) {
+                itsContent.setStem(content.getStem());
+            }
+            if (content.getStemTts() != null) {
+                itsContent.setStemTTS(content.getStemTts());
+            }
+            if (content.getTitle() != null) {
+                itsContent.setTitle(content.getTitle());
+            }
+            if (content.getOptionlist() != null) {
+                itsContent.setOptions(readOptions(document, content.getOptionlist()));
+            }
+            if (content.getKeyboard() != null) {
+                itsContent.setKeyboard(readKeyboard(content.getKeyboard()));
+            }
+            if (content.getApipAccessibility() != null) {
+                itsContent.setApip(readApip(content.getApipAccessibility()));
+            }
+            if (content.getAttachmentlist() != null) {
+                itsContent.setAttachments(readAttachments(document, content.getAttachmentlist().getAttachment()));
+            }
+            processGenericElements(itsContent, content);
+            document.addContent(itsContent);
+        }
+    }
+
+    private ITSQTI readQti(Qti qti) {
+        ITSQTI itsQti = new ITSQTI();
+
+        if (qti.getSpec() != null) {
+            // get the qti specification ("itemBody" or "assessmentItem")
```

Comment:
Same comment here about the null check. What happens if you pass null into the setter below?  If nothing I suggest removing the conditional